go/config/enclave_cli_flags.go

```diff
@@ -0,0 +1,98 @@
+		L2GasLimitFlag:                flag.NewUint64Flag(L2GasLimitFlag, 0, ""),
+		ObscuroGenesisFlag:            flag.NewStringFlag(ObscuroGenesisFlag, "", FlagDescriptionMap[ObscuroGenesisFlag]),
+		L1ChainIDFlag:                 flag.NewInt64Flag(L1ChainIDFlag, 0, FlagDescriptionMap[L1ChainIDFlag]),
+		ObscuroChainIDFlag:            flag.NewInt64Flag(ObscuroChainIDFlag, 0, FlagDescriptionMap[ObscuroChainIDFlag]),
+		UseInMemoryDBFlag:             flag.NewBoolFlag(UseInMemoryDBFlag, false, FlagDescriptionMap[UseInMemoryDBFlag]),
+		ProfilerEnabledFlag:           flag.NewBoolFlag(ProfilerEnabledFlag, false, FlagDescriptionMap[ProfilerEnabledFlag]),
+		DebugNamespaceEnabledFlag:     flag.NewBoolFlag(DebugNamespaceEnabledFlag, false, FlagDescriptionMap[DebugNamespaceEnabledFlag]),
+	}
+}
+
+// enclaveRestrictedFlags are the flags that the enclave can receive ONLY over the Ego signed enclave.json
+var enclaveRestrictedFlags = map[string]*flag.TenFlag{
+	L1ChainIDFlag:             flag.NewInt64Flag(L1ChainIDFlag, 0, FlagDescriptionMap[L1ChainIDFlag]),
+	ObscuroChainIDFlag:        flag.NewInt64Flag(ObscuroChainIDFlag, 0, FlagDescriptionMap[ObscuroChainIDFlag]),
+	ObscuroGenesisFlag:        flag.NewStringFlag(ObscuroGenesisFlag, "", FlagDescriptionMap[ObscuroGenesisFlag]),
+	UseInMemoryDBFlag:         flag.NewBoolFlag(UseInMemoryDBFlag, false, FlagDescriptionMap[UseInMemoryDBFlag]),
+	ProfilerEnabledFlag:       flag.NewBoolFlag(ProfilerEnabledFlag, false, FlagDescriptionMap[ProfilerEnabledFlag]),
+	DebugNamespaceEnabledFlag: flag.NewBoolFlag(DebugNamespaceEnabledFlag, false, FlagDescriptionMap[DebugNamespaceEnabledFlag]),
+}
+
+// Flag names.
+const (
+	HostIDFlag                    = "hostID"
+	HostAddressFlag               = "hostAddress"
+	AddressFlag                   = "address"
+	NodeTypeFlag                  = "nodeType"
+	L1ChainIDFlag                 = "l1ChainID"
+	ObscuroChainIDFlag            = "obscuroChainID"
+	WillAttestFlag                = "willAttest"
+	ValidateL1BlocksFlag          = "validateL1Blocks"
+	ManagementContractAddressFlag = "managementContractAddress"
+	LogLevelFlag                  = "logLevel"
+	LogPathFlag                   = "logPath"
+	UseInMemoryDBFlag             = "useInMemoryDB"
+	EdgelessDBHostFlag            = "edgelessDBHost"
+	SQLiteDBPathFlag              = "sqliteDBPath"
+	ProfilerEnabledFlag           = "profilerEnabled"
+	MinGasPriceFlag               = "minGasPrice"
+	MessageBusAddressFlag         = "messageBusAddress"
+	SequencerIDFlag               = "sequencerID"
+	ObscuroGenesisFlag            = "obscuroGenesis"
+	DebugNamespaceEnabledFlag     = "debugNamespaceEnabled"
+	MaxBatchSizeFlag              = "maxBatchSize"
+	MaxRollupSizeFlag             = "maxRollupSize"
+	L2BaseFeeFlag                 = "l2BaseFee"
+	L2CoinbaseFlag                = "l2Coinbase"
+	L2GasLimitFlag                = "l2GasLimit"
+)
+
+var FlagDescriptionMap = map[string]string{
```

Comment:
is this map needed?
Can't the descriptions live in the EnclaveFlags thing?

Reply:
Fixed !